gobblin-utility/src/main/java/gobblin/util/request_allocation/VectorAlgebra.java

```diff
@@ -0,0 +1,37 @@
+package gobblin.util.request_allocation;
+
+/**
+ * Basic vector operations. These operations are NOT safe (e.g. no bound checks on vectors), so they are package-private.
+ */
+class VectorAlgebra {
+
+  /**
+   * Performs x + cy
+   */
+  static double[] addVector(double[] x, double[] y, double c, double[] reuse) {
+    if (reuse == null) {
```

Comment:
Original content of `reuse` is not used. It is getting all new values no matter `null`. Why not simply remove it from parameter and return as a new array?